packages/common/src/rule-engine/errors.ts

```diff
@@ -14,35 +14,26 @@ export class DataValueNotFoundError extends Error {
   }
 }
 
-export class MissingKeyError extends Error {
-  constructor() {
-    super(`Rule is missing the key field`);
-    this.name = MissingKeyError.name;
-  }
-}
-
 export class ValidationFailedError extends Error {
-  errors:
-    | Array<{
-        message: string;
-        path: string;
-      }>
-    | undefined;
-
-  constructor(key: string, operation: string, error?: ZodError) {
-    super(`Validation failed for '${key}', message: ${operation}`);
-    this.name = 'ValidationFailedError';
-
-    this.errors = error?.errors.map(zodIssue => ({
+  errors: { message: string; path: string }[] | undefined;
+  constructor(key: string, message: string, error?: ZodError) {
+    const errors = error?.errors.map(zodIssue => ({
       message: zodIssue.message,
       path: zodIssue.path.join('.'),
     }));
+
+    super(
+      `Validation failed for '${key}', message: ${message}, ${
+        errors ? `error: ${JSON.stringify(error, null, 2)}` : '  '
+      }`,
+    );
+    this.name = 'ValidationFailedError';
+    this.errors = errors;
```

Comment:
For better logging:
```ts
this.cause = error
```

Reply:
its array of messaged from zodErrors, are you sure cause it should be an array 
`Enumerable | no`

<img width="812" alt="image" src="https://github.com/user-attachments/assets/69d97494-61ef-4c04-a341-7bd73140537b">